### ISSUE TITLE
InaccessibleObjectException on JDK17

### ISSUE THREAD

**Issue by jomu78** [joinfaces/joinfaces#1559]
**Configuration**:

- JDK 17.0.3
- Maven 3.8.6
- Joinfaces 4.7.2
- Example project: https://github.com/joinfaces/joinfaces-maven-jar-example

**Problem**
Application does not start
`Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @7bfcd12c`

**Workaround**
- compile against / start with JDK 11
- start with --add-opens java.base/java.lang=ALL-UNNAMED

**(potentially) Related Bugs**
- #1326
- https://github.com/spring-projects/spring-framework/issues/28699
-

**Comment by schmidti159** [comment]
I had a look into this issue but was not able to find a fix within joinfaces.
What I found: 
- When the proxy for the `FacesContext` bean is created it has the name `$javax.faces.context.FacesContext$$EnhancerBySpringCGLIB$$5ea4f6ed`
- The `$` is added because the class name starts with java (see also https://github.com/cglib/cglib/blame/975e481faf39c91b8ac5b9b3d62822b7c52c5f47/cglib/src/main/java/net/sf/cglib/core/DefaultNamingPolicy.java#L41-L42)
- Because of this `$` the class cannot be created using `java.lang.invoke.MethodHandles$Lookup.defineClass`
It throws the following exception (that is lost and not logged):
```
Caused by: java.lang.IllegalArgumentException: $javax.faces.context.FacesContext$$EnhancerBySpringCGLIB$$674e9b90 not in same package as lookup class
	at java.base/java.lang.invoke.MethodHandleStatics.newIllegalArgumentException(MethodHandleStatics.java:169)
	at java.base/java.lang.invoke.MethodHandles$Lookup$ClassFile.newInstance(MethodHandles.java:2292)
	at java.base/java.lang.invoke.MethodHandles$Lookup.makeClassDefiner(MethodHandles.java:2327)
	at java.base/java.lang.invoke.MethodHandles$Lookup.defineClass(MethodHandles.java:1851)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	... 43 more
```
- When the class cannot be created with `Lookup.defineClass(..)` (the java 9+ compliant way) Spring tries to use the old method `Classloader.defineClass(..)` which throws the `InaccessibleObjectException` for java 17 if `--add-opens` is not used.
 
A similar issue was also discussed here: https://github.com/spring-projects/spring-framework/issues/27622

As the issue (and similar ones) were fixed in Spring Framework 6 I tried to start the example for version 5.0.x and could not reproduce the issue when started with Java 17.
